File: ruby/pe012.rb
#!/usr/bin/env ruby
# http://projecteuler.net/problem=12
def divisors (n)
  result = 2
  (2..Math.sqrt(n)).each do |d|
    if n % d == 0
      result += 2
    end
  end
  result
end

limit = 500
triangle = 0
n = 1
while true
  triangle = (n * (n + 1)) / 2
  d = divisors(triangle)
  break if d > limit
  n += 1
end
puts triangle

File: go/pe012.go
// http://projecteuler.net/problem=12
package main

import (
	"./euler"
	"fmt"
)

func main() {
	generator := euler.Triangle()
	triangle := 0

	for len(euler.Divisors(triangle)) <= 500 {
		triangle = generator()
	}

	fmt.Println(triangle)
}
